--- tclreadline-2.1.0/tclreadline.c.orig	Thu Sep 21 02:44:34 2000
+++ tclreadline-2.1.0/tclreadline.c	Thu Jul 10 19:08:06 2003
@@ -491,7 +491,6 @@
 	TclReadlineTerminate(LINE_EOF);
 
     } else {
-
 	/**
 	 * From version 0.9.3 upwards, all lines are
 	 * returned, even empty lines. (Only non-empty
@@ -501,7 +500,9 @@
 	 */
 
 	char* expansion = (char*) NULL;
+	Tcl_DString result;
 	int status = history_expand(ptr, &expansion);
+	Tcl_DStringInit(&result);
 
 	if (status >= 1) {
 	    /* TODO: make this a valid tcl output */
@@ -514,8 +515,9 @@
 	/**
 	 * TODO: status == 2 ...
 	 */
-
-	Tcl_AppendResult(tclrl_interp, expansion, (char*) NULL);
+	Tcl_ExternalToUtfDString(NULL, expansion, strlen(expansion), &result);
+	Tcl_AppendResult(tclrl_interp, Tcl_DStringValue(&result), (char*) NULL);
+	Tcl_DStringFree(&result);
 
 #ifdef EXECUTING_MACRO_HACK
 	/**
