--- generic/psGdbm.c.orig	2025-04-11 17:33:56.192225899 +0900
+++ generic/psGdbm.c	2025-04-11 17:34:55.851481005 +0900
@@ -144,7 +144,7 @@
      void *handle,
      const char   *key,
      char **dataptrptr,
-     size_t    *lenptr)
+     Tcl_Size  *lenptr)
 {
     GDBM_FILE dbf = (GDBM_FILE)handle;
     datum drec, dkey;
@@ -184,7 +184,7 @@
     void *handle,
     char   **keyptrptr,
     char  **dataptrptr,
-    size_t     *lenptr)
+    Tcl_Size   *lenptr)
 {
     GDBM_FILE dbf = (GDBM_FILE)handle;
     datum drec, dkey;
@@ -225,7 +225,7 @@
     void *handle,
     char   **keyptrptr,
     char  **dataptrptr,
-    size_t     *lenptr)
+    Tcl_Size   *lenptr)
 {
     GDBM_FILE dbf = (GDBM_FILE)handle;
     datum drec, dkey, dnext;
@@ -273,7 +273,7 @@
     void *handle,
     const char   *key,
     char     *dataptr,
-    size_t        len)
+    Tcl_Size      len)
 {
     GDBM_FILE dbf = (GDBM_FILE)handle;
     datum drec, dkey;
