--- tclreadline-2.1.0/tclreadline.c.orig	2004-12-15 13:09:51.000000000 +0900
+++ tclreadline-2.1.0/tclreadline.c	2004-12-15 13:12:53.000000000 +0900
@@ -506,55 +506,34 @@
 	 * empty strings.
 	 */
 
-	char* expansion = (char*) NULL;
 	Tcl_DString result;
-	int status = history_expand(ptr, &expansion);
 	Tcl_DStringInit(&result);
 
-	if (status >= 1) {
-	    /* TODO: make this a valid tcl output */
-	    printf("%s\n", expansion);
-	} else if (-1 == status) {
-	    Tcl_AppendResult
-	    (tclrl_interp, "error in history expansion\n", (char*) NULL);
-	    TclReadlineTerminate(TCL_ERROR);
-	}
-	/**
-	 * TODO: status == 2 ...
-	 */
-	Tcl_ExternalToUtfDString(NULL, expansion, strlen(expansion), &result);
+	Tcl_ExternalToUtfDString(NULL, ptr, strlen(ptr), &result);
 	Tcl_AppendResult(tclrl_interp, Tcl_DStringValue(&result), (char*) NULL);
 	Tcl_DStringFree(&result);
 
-#ifdef EXECUTING_MACRO_HACK
-	/**
-	 * don't stuff macro lines
-	 * into readline's history.
-	 */
-	if(!_rl_executing_macro) {
-#endif
+	{
 	    /**
 	     * don't stuff empty lines
 	     * into readline's history.
 	     * don't stuff twice the same
 	     * line into readline's history.
 	     */
-	    if (expansion && *expansion && (!tclrl_last_line ||
-		    strcmp(tclrl_last_line, expansion))) {
-		add_history(expansion);
+	    if (ptr && *ptr && (!tclrl_last_line ||
+		    strcmp(tclrl_last_line, ptr))) {
+		add_history(ptr);
 	    }
 	    if (tclrl_last_line)
 		free(tclrl_last_line);
-	    tclrl_last_line = strdup(expansion);
-#ifdef EXECUTING_MACRO_HACK
+	    tclrl_last_line = strdup(ptr);
 	}
-#endif
+
 	/**
 	 * tell the calling routines to terminate.
 	 */
 	TclReadlineTerminate(LINE_COMPLETE);
 	FREE(ptr);
-	FREE(expansion);
     }
 }
 
@@ -672,28 +651,6 @@
     int status;
     rl_completion_append_character = ' '; /* reset, just in case ... */
 
-    if (text && ('!' == text[0]
-	    || (start && rl_line_buffer[start - 1] == '!' /* for '$' */))) {
-	char* expansion = (char*) NULL;
-	int oldlen = strlen(rl_line_buffer);
-	status = history_expand(rl_line_buffer, &expansion);
-	if (status >= 1) {
-	    rl_extend_line_buffer(strlen(expansion) + 1);
-	    strcpy(rl_line_buffer, expansion);
-	    rl_end = strlen(expansion);
-	    rl_point += strlen(expansion) - oldlen;
-	    FREE(expansion);
-	    /*
-	     * TODO:
-	     * because we return 0 == matches,
-	     * the filename completer will still beep.
-	     rl_inhibit_completion = 1;
-	     */
-	    return matches;
-	}
-	FREE(expansion);
-    }
-
     if (tclrl_custom_completer) {
 	char start_s[BUFSIZ], end_s[BUFSIZ];
 	Tcl_Obj* obj;
