--- tclreadline-2.1.0/tclreadlineCompleter.tcl.orig	2014-02-07 14:56:50.644307874 +0900
+++ tclreadline-2.1.0/tclreadlineCompleter.tcl	2014-02-07 15:54:09.992331190 +0900
@@ -282,7 +282,7 @@
 
 	set executables 0
 	foreach dir [split $env(PATH) :] {
-		if {[catch [list set files [glob -nocomplain ${dir}/*]]]} { continue }
+		if {[catch {set files [glob -nocomplain ${dir}/*]}]} { continue }
 		foreach file $files {
 			if {[file executable $file]} {
 				lappend executables [file tail ${file}]
@@ -464,7 +464,7 @@
 	# because cget will not return the
 	# option table.
 	#
-	if {[catch [list set option_table [eval ${cmd} configure]] msg]} {
+	if {[catch {set option_table [eval ${cmd} configure]} msg]} {
 		return ""
 	}
 	set classes ""
@@ -491,7 +491,7 @@
 	# because cget will not return the
 	# option table.
 	#
-	if {[catch [list set option_table [eval ${cmd} configure]] msg]} {
+	if {[catch {set option_table [eval ${cmd} configure]} msg]} {
 		return 0
 	}
 	set retval 0
@@ -610,7 +610,7 @@
 	if {"." == [string index ${line} 0]} {
 		# it's a widget. Try to get it's class name.
 		#
-		if {![catch [list set class [winfo class [Lindex ${line} 0]]]]} {
+		if {![catch {set class [winfo class [Lindex ${line} 0]]}]} {
 			if {[string length [info proc ${class}Obj]]} {
 				set result [${class}Obj ${text} ${start} ${end} ${line} ${pos}]
 				# puts stderr result=|$result|
@@ -625,7 +625,7 @@
 			}
 		}
 	}
-	if {![catch [list set type [image type ${cmd}]]]} {
+	if {![catch {set type [image type ${cmd}]}]} {
 		switch -- ${type} {
 			photo {
 				set result [PhotoObj ${text} ${start} ${end} ${line} ${pos}]
@@ -1021,13 +1021,13 @@
 # if the index is `end' or == [Llength $line].
 #
 proc Lindex {line pos} {
-	if {[catch [list set sub [lindex ${line} ${pos}]]]} {
+	if {[catch {set sub [lindex ${line} ${pos}]}]} {
 		if {"end" == ${pos} || [Llength ${line}] == ${pos}} {
 			return [IncompleteListRemainder ${line}]
 		}
 		set line [ProperList ${line}]
 		# puts stderr \nproper_line=|$proper_line|
-		if {[catch [list set sub [lindex ${line} ${pos}]]]} { return {} }
+		if {[catch {set sub [lindex ${line} ${pos}]}]} { return {} }
 	}
 	return ${sub}
 }
@@ -1036,9 +1036,9 @@
 # save `llength' (see above).
 #
 proc Llength {line} {
-	if {[catch [list set len [llength ${line}]]]} {
+	if {[catch {set len [llength ${line}]}]} {
 		set line [ProperList ${line}]
-		if {[catch [list set len [llength ${line}]]]} { return {} }
+		if {[catch {set len [llength ${line}]}]} { return {} }
 	}
 	# puts stderr \nline=$line
 	return ${len}
@@ -1048,10 +1048,10 @@
 # save `lrange' (see above).
 #
 proc Lrange {line first last} {
-	if {[catch [list set range [lrange ${line} ${first} ${last}]]]} {
+	if {[catch {set range [lrange ${line} ${first} ${last}]}]} {
 		set rest [IncompleteListRemainder ${line}]
 		set proper [ProperList ${line}]
-		if {[catch [list set range [lindex ${proper} ${first} ${last}]]]} {
+		if {[catch {set range [lindex ${proper} ${first} ${last}]}]} {
 			return {}
 		}
 		if {"end" == ${last} || [Llength ${line}] == ${last}} {
@@ -1346,7 +1346,7 @@
 	set cmd [string range ${cmd} 1 [expr [string length ${cmd}] - 2]]
 	set rest [String range ${line} [expr ${idx} + 1] end]
 
-	if {[catch [list set result [string trim [eval ${cmd}]]]]} {return}
+	if {[catch {set result [string trim [eval ${cmd}]]}]} {return}
 
 	set line ${result}${rest}
 	set diff [expr [string length ${result}] - ([string length ${cmd}] + 2)]
@@ -1460,7 +1460,7 @@
 			# this can raise an error, if alias is
 			# no valid command.
 			#
-			if {[catch [list set alias [namespace origin $alias]]]} {
+			if {[catch {set alias [namespace origin $alias]}]} {
 				return ""
 			}
 
@@ -1496,9 +1496,9 @@
 				set mod [StripPrefix $part]
 				# puts stderr mod=$mod
 
-				if {[catch [list set script_result \
+				if {[catch {set script_result \
 					[namespace eval ::tclreadline::${namespc} \
-					[list complete(${cmd}) $part $start $end $line $pos $mod]]]\
+					[list complete(${cmd}) $part $start $end $line $pos $mod]]}\
 					::tclreadline::errorMsg]
 				} {
 					error [list error during evaluation of `complete(${cmd})']
@@ -2889,7 +2889,7 @@
 		}
 		3 {
 			set versions ""
-			catch [list set versions [package versions [Lindex ${line} 2]]]
+			catch {set versions [package versions [Lindex ${line} 2]]}
 			switch -- ${cmd} {
 				forget {}
 				ifneeded {
@@ -3211,8 +3211,8 @@
 		2 {
 			if {${text} == "" || ${text} == "\"" || ${text} == "\{"} {
 				# set line [QuoteQuotes $line]
-				if {[catch [list set value [list [uplevel [info level] \
-					set [Lindex ${line} 1]]]] msg]
+				if {[catch {set value [list [uplevel [info level] \
+					set [Lindex ${line} 1]]]} msg]
 				} {
 					return ""
 				} else {
@@ -3706,8 +3706,8 @@
 		0 {
 			if {$text == "" || $text == "\"" || $text == "\{"} {
 				set line [QuoteQuotes $line]
-				if {[catch [list set value [list [uplevel [info level] \
-					set [PreviousWord $start $line]]]] msg]
+				if {[catch {set value [list [uplevel [info level] \
+					set [PreviousWord $start $line]]]} msg]
 				} {
 					return ""
 				} else {
